src/lib.rs

```diff
@@ -1139,41 +1144,49 @@ impl BlazeSymbolizer {
                     return vec![];
                 };
 
-                let sym = resolver.find_symbol(*addr);
+                let res_syms = resolver.find_symbols(*addr);
                 let linfo = if self.line_number_info {
                     resolver.find_line_info(*addr)
                 } else {
                     None
                 };
-                if sym.is_none() && linfo.is_none() {
-                    vec![]
-                } else if sym.is_none() {
-                    let linfo = linfo.unwrap();
-                    vec![SymbolizedResult {
-                        symbol: "".to_string(),
-                        start_address: 0,
-                        path: linfo.path,
-                        line_no: linfo.line_no,
-                        column: linfo.column,
-                    }]
-                } else if let Some(linfo) = linfo {
-                    let (sym, start) = sym.unwrap();
-                    vec![SymbolizedResult {
-                        symbol: String::from(sym),
-                        start_address: start,
-                        path: linfo.path,
-                        line_no: linfo.line_no,
-                        column: linfo.column,
-                    }]
+                if res_syms.is_empty() {
+                    if linfo.is_some() {
+                        let linfo = linfo.unwrap();
+                        vec![SymbolizedResult {
+                            symbol: "".to_string(),
+                            start_address: 0,
```

Comment:
I don't think we should use any sentinel values like this. If there is the potential for data not being present then we should use `Option` instead to make this explicit, in my opinion. Behavior like this is precisely where problems such as https://github.com/libbpf/blazesym/issues/15 are rooted in.

Reply:
Not really.  #15 is about multiple symbols with the same address, but we returned only one of them.
However, I agree that the `Option` is better here.